### ISSUE TITLE
Can Proprerty "audio-codec" provide detail information about DTS?

### ISSUE THREAD

**Issue by macdavis** [mpv-player/mpv#4569]
### mpv version and platform
mpv git 41b3b11
MacOS 10.12.5 (16F73)

### Reproduction steps
Play any DTS file with [mpv-stats scripts ](https://github.com/Argon-/mpv-stats/)
### Expected behavior
Display more information about DTS, such as DTS-HD Master Audio, DTS-HD High Resolution Audio, DTS 96/24, DTS-ES, etc.
### Actual behavior
Display generic information only, `dca (DCA (DTS Coherent Acoustics))`
### Log file
 [Log file](http://sprunge.us/jFej)

### Sample files

[Sample files](https://0x0.st/02v.mka)

**Comment by wiiaboo** [comment]
Could be obtained from FFmpeg, by retrieving the profile name of the codec with [`avcodec_profile_name`](https://www.ffmpeg.org/doxygen/trunk/group__lavc__misc.html#ga7f9d1f2fbf0978fc1ddc2fe196f29fa4).

**Comment by wiiaboo** [comment]
Though not sure how exactly, since codec->profile always shows -99 (FF_PROFILE_UNKNOWN) for me.
```diff --git a/demux/demux_lavf.c b/demux/demux_lavf.c
index da0dbd46a..8a62099d4 100644
--- a/demux/demux_lavf.c
+++ b/demux/demux_lavf.c
@@ -707,6 +707,7 @@ static void handle_new_stream(demuxer_t *demuxer, int i)
         sh->ff_index = st->index;
         sh->codec->codec = mp_codec_from_av_codec_id(codec->codec_id);
         sh->codec->codec_tag = codec->codec_tag;
+        MP_INFO(demuxer, "profile: %i\n", codec->profile);
         sh->codec->lav_codecpar = avcodec_parameters_alloc();
         if (sh->codec->lav_codecpar)
             avcodec_parameters_copy(sh->codec->lav_codecpar, codec);
```

**Comment by wm4** [comment]
Also works only for demux_lavf.

ad_spdif.c has some code to attempt to extract the profile (since it actually matters there), but it remains messy.

**Comment by macdavis** [comment]
@wiiaboo 

I saw this in mpv/video/decode/vd_lavc.c. It successfully gets the profile information with hardware decoder.


> static enum AVPixelFormat get_format_hwdec(struct AVCodecContext *avctx, const enum AVPixelFormat *fmt)
> ...
> const char *profile = avcodec_profile_name(avctx->codec_id, avctx->profile);
> MP_VERBOSE(vd, "Codec profile: %s (0x%x)\n", profile ? profile : "unknown", avctx->profile);


Is it possible to apply to audio decoder?

**Comment by wm4** [comment]
This is for hwaccels, which use the same code as the software decoder to parse the data before passing it to the hardware decoder.

**Comment by macdavis** [comment]
I understand that. What I mean is, how to insert this function `const char *profile = avcodec_profile_name` somewhere in the audio decoder to parse the profile.

**Comment by wm4** [comment]
You mean basically for debugging?

**Comment by macdavis** [comment]
Not really. Ultimately, I wish I can see different types of DTS audio distinguished when using "audio-codec" property. I watch a lot of blu-rays. Most of them are encoded in DTS-HD Master Audio. But currently Lua script can only print  "dca (DCA (DTS Coherent Acoustics))"

**Comment by kasper93** [comment]
Fixed by #13841 and  #13906

See [`track-list/N/codec-profile`](https://mpv.io/manual/master/#command-interface-track-list/n/codec-profile)
